ros2_batch_job/__main__.py

```diff
@@ -319,14 +320,23 @@ def build_and_test(args, job):
     if ret_build:
         return ret_build
 
+    if args.enable_sanitizer_type:
+        ret_build_with_sanitizer = build_with_sanitizer(cmd, args, job)
```

Comment:
How much overhead does building the selected packages a second introduce? It would be better to build them with the desired flags in the first pass.

Reply:
Good point, I was experimenting the same while working on thread sanitizer job. I have modified the build command to perform the build in three steps: build till regex packages, build regex packages with sanitizer, build above regex packages. This avoids the additional pass of colcon build on all packages